### PR TITLE
Added Validator for Discord API credentials

### PR DESCRIPTION
**Pull request by Bpendragon**

### CODE REVIEW

File: Src/Plugins/Tests.Security/Tests.Security.csproj [hunk 1]
```diff
@@ -13,6 +13,11 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <ItemGroup>
+    <None Remove="TestData\SecurePlaintextSecrets\ExpectedOutputs\SEC101_046.DiscordApiCredentials.sarif" />
```
Comment (eddynaka) [inline Src/Plugins/Tests.Security/Tests.Security.csproj:17]: >Remove [](http://example.com/codeflow?start=10&length=6)

pls, remove

File: Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                return;
+            }
+
+            string message = null;
+            ResultLevelKind resultLevelKind = default;
+            var fingerprint = new Fingerprint(fingerprintText);
+            var keyValuePairs = new Dictionary<string, string>();
+
+            DiscordApiCredentialsValidator.IsValidDynamic(ref fingerprint,
+                                               ref message,
```
Comment (eddynaka) [inline Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs:29]: >ref [](http://example.com/codeflow?start=47&length=3)

pls, align by ref fingerprint (check other files to see the pattern)

File: Src/Sarif.PatternMatcher.Sdk/AssetPlatform.cs [hunk 3]
```diff
@@ -40,6 +40,11 @@ public enum AssetPlatform
         /// </summary>
         Cloudant,
 
+        /// <summary>
+        /// Discord Platform. https://discord.com/developers
```
Comment (eddynaka) [inline Src/Sarif.PatternMatcher.Sdk/AssetPlatform.cs:44]: >P [](http://example.com/codeflow?start=20&length=1)

nit: lowercase p

File: Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs (hunk 2 above)
Comment (Bpendragon) [reply Src/Plugins/Tests.Security/Validators/SEC101_046.DiscordApiCredentialsValidatorTests.cs]: fixed

File: Src/Sarif.PatternMatcher.Sdk/AssetPlatform.cs (hunk 3 above)
Comment (Bpendragon) [reply Src/Sarif.PatternMatcher.Sdk/AssetPlatform.cs]: fixed

File: Src/Plugins/Tests.Security/Tests.Security.csproj (hunk 1 above)
Comment (Bpendragon) [reply Src/Plugins/Tests.Security/Tests.Security.csproj]: removed

**Review by eddynaka — APPROVED** [review]

**Pull request merged**